### PR TITLE
ceph-disk: workaround gperftool hang

### PR DESCRIPTION
**Pull request by dachary**
Temporary workaround: if ceph-osd --mkfs does not
complete within 5 minutes, assume it is blocked
because of https://github.com/gperftools/gperftools/issues/786

References http://tracker.ceph.com/issues/13522

Signed-off-by: Loic Dachary <loic@dachary.org>

### CODE REVIEW

**Comment by dachary** [comment]
test this please

**Comment by dachary** [comment]
test this please

**Comment by tchaikov** [comment]
retest this please

**Comment by dachary** [comment]
@tchaikov I manually tested it to work on Ubuntu 14.04. It shows that there is no regression. I did not test what happens when it fails, I realize that I should setup the conditions for that to happen. Maybe with a fixture of ceph-osd that fails the first time and succeed the rest of the time. And add that to the ceph-disk suite.

**Comment by dachary** [comment]
@tchaikov actually, the test can be done within make check, no need for more than one machine. I'll do that.

File: src/ceph-disk/ceph_disk/main.py [hunk 1]
```diff
@@ -2642,6 +2642,35 @@ def populate_data_path(self, path, *to_prepare_list):
...
+# when http://tracker.ceph.com/issues/13522 is fixed
+#
+def ceph_osd_mkfs(arguments):
+    timeout = _get_command_executable(['timeout', '300'])
+    mkfs_ok = False
+    error = 'unknown error'
+    for i in (1, 2, 3, 4, 5):
+        try:
+            _check_output(timeout + arguments)
+            mkfs_ok = True
```
Comment (tchaikov) [inline src/ceph-disk/ceph_disk/main.py]: could return right away.

**Comment by tchaikov** [comment]
lgtm @dachary do you still plan to add a test for the case where `ceph-osd --mkfs` fails?

**Comment by dachary** [comment]
@tchaikov yes, verifying it. It makes me nervous to not cover the case where it's suppose to fix things ... Thanks a lot for your review.

File: src/ceph-disk/ceph_disk/main.py [hunk 2]
```diff
@@ -2642,6 +2642,35 @@ def populate_data_path(self, path, *to_prepare_list):
...
+# and retry a few times.
+#
+# Remove this function calls with command_check_call
+# when http://tracker.ceph.com/issues/13522 is fixed
+#
+def ceph_osd_mkfs(arguments):
+    timeout = _get_command_executable(['timeout', '300'])
+    mkfs_ok = False
+    error = 'unknown error'
+    for i in (1, 2, 3, 4, 5):
```
Comment (tchaikov) [inline src/ceph-disk/ceph_disk/main.py]: nit: `range(5)` =)

**Comment by dachary** [comment]
test this please (mount bluefs crash)

**Pull request merged**